### DISCUSSION TITLE
A way to properly change a default route

### DISCUSSION THREAD

**Discussion by stratboy** [vercel/next.js#21322] [Help]
Hi, I've got a page called service.js, so I also have an auto-route like `/services`

I want that page to be accessible from `/servizi`

So I opened my `next.config.js` file and wrote:

``` js
module.exports = {  
  async rewrites() {
    return [
      {
        source: '/servizi',
        destination: '/services',
      },
    ]
  },
}
```

Works fine, except that now that resource is reachable both from `/servizi` AND `/services`. What I'd like though, is that the resource would be accessible only from `/servizi`.

How to?

**Comment by jpedroschmitz** [comment]
Rewrites are not able to override public files or routes in the `pages` directory as these have higher priority than rewrites. 

The only option here it's to change the filename to `servizi.js` and add a redirect from `/services` to `/servizi`.

```js
module.exports = {
  async redirects() {
    return [
      {
        source: '/services',
        destination: '/servizi',
        permanent: true,
      },
    ]
  },
}
```

[Rewrites docs](https://nextjs.org/docs/api-reference/next.config.js/rewrites).

**Reply by stratboy** [reply]
Ok thank you. Then, I guess languages will have their own rewrite mechanics... Because languages definitely need to have completely different routes despite filenames in /pages.
